include/s3select.h

```diff
@@ -2132,49 +2142,89 @@ class base_s3object
 	return m_returned_bytes_size;
   }
 
-  void result_values_to_string(multi_values& projections_resuls, std::string& result)
+  void json_result_format(multi_values& projections_results, std::string& result, std::string& output_delimiter, std::string& last_key)
   {
-    size_t i = 0;
+    result += "{";
+    for (size_t i = 0; i < projections_results.values.size(); ++i)
+    {
+      auto& res = projections_results.values[i];
+      std::string label = "_";
+      label += std::to_string(i + 1);
+
+      if (i > 0) {
+        result += output_delimiter;
+      }
+          
+      if(!is_star && m_aggr_flow)  {
+        result += "\"" + label + "\":";
+        } else if(!is_star && !m_aggr_flow) {
+          result += "\"" + last_key + "\":";
+      }
+
+      result.append(res->to_string());
+      m_returned_bytes_size += strlen(res->to_string());
+      }
+    result += "}";
+  }
+
+
+  void result_values_to_string(multi_values& projections_resuls, std::string& result)
+{   
+    std::string last_key;
     std::string output_delimiter(1,m_csv_defintion.output_column_delimiter);
     std::string output_row_delimiter(1,m_csv_defintion.output_row_delimiter);
+    std::vector<std::string> key_path(m_sa->get_json_key_path());
+    
+    if(key_path.size()) {
+      last_key = key_path[key_path.size()-1];
+    }
 
-    for(auto& res : projections_resuls.values)
-    {
-	    if(fp_ext_debug_mesg)
-		      fp_ext_debug_mesg( res->to_string() );
-
-            if (m_csv_defintion.quote_fields_always) {
-              std::ostringstream quoted_result;
-              quoted_result << std::quoted(res->to_string(),m_csv_defintion.output_quot_char, m_csv_defintion.escape_char);
-              result.append(quoted_result.str());
-	      m_returned_bytes_size += quoted_result.str().size();
-            }//TODO to add asneeded
-	    else
-	    {
-            	result.append(res->to_string());
-		m_returned_bytes_size += strlen(res->to_string());
-	    }
-
-            if(!m_csv_defintion.redundant_column) {
-              if(++i < projections_resuls.values.size()) {
-                result.append(output_delimiter);
-		m_returned_bytes_size += output_delimiter.size();
+    if(is_json)  {
+      json_result_format(projections_resuls, result, output_delimiter, last_key);
```

Comment:
the JSON output format is an indication passing by the AWS/CLI.
the s3select must have an interface to accept that, the same as CSV definitions.
please add this indication (JSON output format) to `s3select_example`.

Reply:
please add the json-output-format indication to `s3select_example`
you can use environment variables.